src/Xamarin.Android.Build.Tasks/Tasks/JavaDependencyVerification.cs

```diff
@@ -370,59 +368,59 @@ public class NuGetPackageVersionFinder
 		}
 	}
 
-	public Artifact? GetJavaInformation (string library, string version, TaskLoggingHelper log)
+	public List<Artifact> GetArtifactsInNugetPackage (string library, string version, TaskLoggingHelper log)
 	{
-		// Check if we already have this one in the cache
-		var dictionary_key = $"{library.ToLowerInvariant ()}:{version}";
-
-		if (cache.TryGetValue (dictionary_key, out var artifact))
-			return artifact;
+		var artifacts = new List<Artifact> ();
 
 		// Find the LockFileLibrary
 		var nuget = lock_file.GetLibrary (library, new NuGet.Versioning.NuGetVersion (version));
 
 		if (nuget is null) {
 			log.LogCodedError ("XA4248", Properties.Resources.XA4248, library, version);
-			return null;
+			return artifacts;
 		}
 
 		foreach (var path in lock_file.PackageFolders)
-			if (CheckFilePath (path.Path, nuget) is Artifact art) {
-				cache.Add (dictionary_key, art);
-				return art;
-			}
+			AddArtifactsFromNuspec (artifacts, path.Path, nuget);
 
-		return null;
+		return artifacts;
 	}
 
-	Artifact? CheckFilePath (string nugetPackagePath, LockFileLibrary package)
+	void AddArtifactsFromNuspec (List<Artifact> artifacts, string nugetPackagePath, LockFileLibrary package)
 	{
 		// Check NuGet tags
 		var nuspec = package.Files.FirstOrDefault (f => f.EndsWith (".nuspec", StringComparison.OrdinalIgnoreCase));
 
 		if (nuspec is null)
-			return null;
+			return;
 
 		nuspec = Path.Combine (nugetPackagePath, package.Path, nuspec);
 
 		if (!File.Exists (nuspec))
-			return null;
+			return;
 
 		var reader = new NuGet.Packaging.NuspecReader (nuspec);
 		var tags = reader.GetTags ();
 
 		// Try the first tag format
-		var match = tag.Match (tags);
+		AddMatchesToCollection (artifacts, tag.Matches (tags));
```

Comment:
While reading up on Regex docs, I came across [this guidance](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expressions#regular-expression-examples):

> When using [System.Text.RegularExpressions](https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions) to process untrusted input, pass a timeout. A malicious user can provide input to RegularExpressions, causing a [Denial-of-Service attack](https://www.cisa.gov/news-events/news/understanding-denial-service-attacks). ASP.NET Core framework APIs that use RegularExpressions pass a timeout.

This *is* untrusted input -- it's from a NuGet package! -- so we should probably be using the [`Regex.Matches(string, string, RegexOptions, TimeSpan)` method](https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.matches?view=net-8.0#system-text-regularexpressions-regex-matches(system-string-system-string-system-text-regularexpressions-regexoptions-system-timespan)).